test/famfs_unit.cpp

```diff
@@ -29,13 +29,40 @@ extern "C" {
 #define FAMFS_MPT "/mnt/famfs"
 #define DIRPATH   "/mnt/famfs/testdir42"
 #define TESTFILE  "/mnt/famfs/testdir42/testfile0"
+#define PATH	  256
+#define SYS_UUID_DIR "/opt/famfs"
 
 TEST(famfs, dummy)
 {
 	printf("Dummy test\n");
 	ASSERT_EQ(0, 0);
 }
 
+TEST(famfs, famfs_create_sys_uuid_file)
+{
+	char sys_uuid_file[PATH];
+	int rc, fd;
+
+	// Check with correct file name and path
+	snprintf(sys_uuid_file, PATH, "/opt/famfs/system_uuid");
+	rc = famfs_create_sys_uuid_file(sys_uuid_file);
+	ASSERT_EQ(rc, 0);
+	// Pass a directory
+	system("mkdir -p /opt/famfs");
+	snprintf(sys_uuid_file, PATH, "%s", SYS_UUID_DIR);
+	rc = famfs_create_sys_uuid_file(sys_uuid_file);
+	ASSERT_NE(rc, 0);
+
+	// create a uuid file
+	fd = open("/tmp/system_uuid", O_RDWR | O_CREAT, 0666);
+	ASSERT_GT(fd, 0);
+	snprintf(sys_uuid_file, PATH, "/tmp/system_uuid");
+	rc = famfs_create_sys_uuid_file(sys_uuid_file);
+	ASSERT_EQ(rc, 0);
+	system("rm /tmp/system_uuid");
+
```

Comment:
This is a good start, but the coverage is not yet sufficient IMO. If you add this makefile target:
```
unit_coverage:	coverage
	sudo rm	-rf coverage
	cd coverage; sudo make famfs_unit_coverage
```
And then look at the unit test coverage, of famfs_create_sys_uuid_file() and famfs_get_system_uuid(), it's not that good. Let's find a way to increase the unit test coverage of both of those functions this PR (which IMO should call both functions and try to cause every branch if possible. (there aren't that many branches...)

Below is the coverage of famfs_create_sys_uuid_file(). Mind you I'm responsible for the sketchy coverage of famfs_get_system_uuid(), but let's address that here...

![Screenshot 2024-03-13 at 10 06 25 AM](https://github.com/cxl-micron-reskit/famfs/assets/6696890/2632cc3d-98a9-477e-9d76-a1822ca152b3)

Reply:
Thanks John. I ran the coverage tests and indeed we could improve here.
I have added / modified tests so that more cases/code is covered.
But, there are a couple of error paths which we also discussed yesterday, where it is difficult to branch into. For example all the tests / commands/ unit tests will be running under sudo and certain error handling cases like failure of mkdir or write() is not easy to fail. 
So based on existing precedence I have introduced (rather reluctantly) a "int mock_uuid" variable to excercise these paths, which has increased the coverage. 

Below is the new coverage for both functions famfs_get_system_uuid() and famfs_create_sys_uuid().

Reply:
![famfs-create-sys_uuid_file](https://github.com/cxl-micron-reskit/famfs/assets/20333180/92322a69-17d4-40e1-80b5-28aedb9075a1)
famfs_create_sys_uuid()

Reply:
![famfs-get-system-uuid](https://github.com/cxl-micron-reskit/famfs/assets/20333180/c3e57da8-632c-4331-bbbb-7b3d41e470de)

famfs_get_system_uuid()

Reply:
This is good for now. We will need to resort to function mocking to get more, I think. We don't need to go there in this PR...